apps/frontend/src/middleware.ts

```
@@ -23,33 +92,54 @@ async function verifyToken(token: string): Promise<boolean> {
-      }
-    }
-    return NextResponse.next()
+  // If !result and trying to access protected routes
+  if (!result && !publicPath.includes(path)) {
+    return NextResponse.redirect(new URL("/", request.url))
+  }
+
+  // If result, valid user  with a verified profile, trying to access a public path
+  if (
+    result &&
+    result.isValidUser &&
```

Biome

[error] Change to an optional chain.

Unsafe fix: Change to an optional chain.


(lint/complexity/useOptionalChain)